backend/src/schema/types/schema.gql

```diff
@@ -32,9 +32,9 @@ type Mutation {
   # Unshout the given Type and ID
   unshout(id: ID!, type: ShoutTypeEnum): Boolean!
   # Follow the given Type and ID
-  follow(id: ID!, type: FollowTypeEnum): Boolean!
+  follow(id: ID!, type: FollowTypeEnum): User
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-18T22:06:20Z" title="Thursday, September 19th 2019, 12:06:20 am +02:00">Sep 19, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -32,9 +32,9 @@ type Mutation {
   # Unshout the given Type and ID
   unshout(id: ID!, type: ShoutTypeEnum): Boolean!
   # Follow the given Type and ID
-  follow(id: ID!, type: FollowTypeEnum): Boolean!
+  follow(id: ID!, type: FollowTypeEnum): User
```


:+1: nice, looks great!

If you need any help or another review, feel free to ask for it @vbelolapotkov.

Reply:
> [<img alt="vbelolapotkov" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/vbelolapotkov) **Authored by [vbelolapotkov](https://github.com/vbelolapotkov)**
_<time datetime="2019-09-19T07:21:07Z" title="Thursday, September 19th 2019, 9:21:07 am +02:00">Sep 19, 2019</time>_

---


@roschaefer actually this looks a little bit weird, especially `FollowTypeEnum` argument. I suppose this was done to reuse the mutation in the future to allow following other entities like posts or something else.

In that case, I'd expect some new `Followable` type (e.g. union type) or interface is returned. What do you think?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-19T18:04:48Z" title="Thursday, September 19th 2019, 8:04:48 pm +02:00">Sep 19, 2019</time>_

---


This makes no sense at all. I guess the `FollowTypeEnum` was supposed to include `Organization`s as well. Something that we decided to not implement any soon. @vbelolapotkov do you want to land another quick contribution and remove `FollowTypeEnum`?

Reply:
> [<img alt="vbelolapotkov" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/vbelolapotkov) **Authored by [vbelolapotkov](https://github.com/vbelolapotkov)**
_<time datetime="2019-09-19T19:41:49Z" title="Thursday, September 19th 2019, 9:41:49 pm +02:00">Sep 19, 2019</time>_

---


@roschaefer will be happy to help here as well. I'd rename the mutation as well to `(un)followUser`

Reply:
> [<img alt="vbelolapotkov" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/vbelolapotkov) **Authored by [vbelolapotkov](https://github.com/vbelolapotkov)**
_<time datetime="2019-09-20T14:01:42Z" title="Friday, September 20th 2019, 4:01:42 pm +02:00">Sep 20, 2019</time>_

---


@roschaefer here it is #1660 , please review when you have time